code/game/objects/items/stacks/stack.dm

```diff
@@ -200,7 +200,10 @@
 				return
 			if(R.time)
 				usr.visible_message("<span class='notice'>[usr] starts building \a [R.title].</span>", "<span class='notice'>You start building \a [R.title]...</span>")
-				if (!do_after(usr, R.time, target = usr))
+				if(HAS_TRAIT(usr.mind, TRAIT_CONSTRUCTION_ADEPT))
+					if(!do_after(usr, (R.time / 2), target = usr))
+						return
+				else if(!do_after(usr, R.time, target = usr))
 					return
```

Comment:
```suggestion
				if(!do_after(usr, R.time * (HAS_TRAIT(usr.mind, TRAIT_CONSTRUCTION_ADEPT) ? 0.5 : 1), target = usr))
					return
```

Reply:
Does DM handle precedence correctly here? Might be a good idea to wrap the ternary in parentheses if you're not 100% sure.

Reply:
Tested it in game. It works fine.

Reply:
Fikou's syntax looks much clearer here